iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -446,29 +439,105 @@ TEST_F(PortManager_test, ConditionVariablesDestroy)
     }
 }
 
+TEST_F(PortManager_test, NodeDataOverflow)
+{
+    std::string process = "Process";
+    std::string node = "Node";
+
+    // first acquire all possible NodeData
+    for (unsigned int i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+        iox::NodeName_t newNodeName(iox::cxx::TruncateToCapacity, node + std::to_string(i));
+        auto nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
+        EXPECT_THAT(nodeData.has_error(), Eq(false));
+    }
+
+    // test if overflow errors get hit
+    {
+        auto errorHandlerCalled{false};
+        auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+            [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+                errorHandlerCalled = true;
+            });
+
+        auto nodeData = m_portManager->acquireNodeData("AnotherProcess", "AnotherNode");
+        EXPECT_THAT(nodeData.has_error(), Eq(true));
+        EXPECT_THAT(errorHandlerCalled, Eq(true));
+        EXPECT_THAT(nodeData.get_error(), Eq(PortPoolError::NODE_DATA_LIST_FULL));
+    }
+
+    // delete one and add one NodeData should be possible now
+    {
+        unsigned int testi = 0U;
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(testi));
+        iox::NodeName_t newNodeName(iox::cxx::TruncateToCapacity, node + std::to_string(testi));
+        m_portManager->deletePortsOfProcess(newProcessName);
+
+        auto nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
+        EXPECT_THAT(nodeData.has_error(), Eq(false));
+        EXPECT_THAT(nodeData.value()->m_node, StrEq(newNodeName));
+    }
+}
+
+TEST_F(PortManager_test, NodeDataDestroy)
+{
+    iox::ProcessName_t process = "Humuhumunukunukuapua'a";
+    iox::NodeName_t node = "Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu";
+    std::vector<iox::runtime::NodeData*> nodeContainer;
+
+    // first acquire all possible NodeData
+    for (unsigned int i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        auto nodeData = m_portManager->acquireNodeData(process, node);
+        EXPECT_THAT(nodeData.has_error(), Eq(false));
+        nodeContainer.push_back(nodeData.value());
+    }
+
+    // so now no NodeData should be available
+    {
+        auto nodeData = m_portManager->acquireNodeData(process, node);
+        EXPECT_THAT(nodeData.has_error(), Eq(true));
+    }
+
+    // set the destroy flag and let the discovery loop take care
+    for (unsigned int i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        nodeContainer[i]->m_toBeDestroyed.store(true, std::memory_order_relaxed);
+    }
+    m_portManager->doDiscovery();
+    nodeContainer.clear(); // These pointers are dangling now
```

Comment:
Remove the comment or expand on this (dangling is only relevant if we try to access them, which we do not). We can always safely forget pointers no one intends to read from if someone else does the cleanup of the data they point to (doDiscovery in this case).

The comment is not helpful to explain this by itself.